src/libraries/Common/src/Interop/Windows/WinHttp/Interop.winhttp_types.cs

```diff
@@ -129,8 +129,10 @@ internal partial class WinHttp
         public const uint WINHTTP_AUTH_TARGET_PROXY = 0x00000001;
 
         public const uint WINHTTP_OPTION_USERNAME = 0x1000;
+        // [SuppressMessage("Microsoft.Security", "CS002:SecretInNextLine", Justification="It is property descriptor, not secret value.")]
         public const uint WINHTTP_OPTION_PASSWORD = 0x1001;
         public const uint WINHTTP_OPTION_PROXY_USERNAME = 0x1002;
+        // [SuppressMessage("Microsoft.Security", "CS002:SecretInNextLine", Justification="It is property descriptor, not secret value.")]
```

Comment:
Why are these commented out?  Shouldn't they either be deleted or uncommented?  Or does the tooling pay attention to commented out SuppressMessage attributes?

Reply:
Yes, it is official way how to suppress report for the next line.

Reply:
I wonder how we will remember to not remove these as dead code! Would this also be recognized?
```c#
// [SuppressMessage("Microsoft.Security", "CS002:SecretInNextLine", Justification="It is property descriptor, not secret value.")]      // Commented line recognized by credscan tool
```
?

Reply:
My preference would be actually running the analyzer in dotnet/runtime.

Reply:
Can you clarify, I thought that's what @aik-jahoda is doing.

Reply:
It doesn't run as part of ci or automatically in local builds. If a dev makes a change, we won't know about violations until after it's merged.

Reply:
CI would be limited value, as a leak has already occurred. 
In my mind this tool is mainly useful for closed repos. In public repos we need auto revoking, which Github provides as you know (at least for some token types).
As for locally a build step would not catch inadvertent adds - you really need a git hook.

Reply:
> CI would be limited value, as a leak has already occurred.

None of the things being flagged are problematic, yet there's dev work required to suppress them (and a lot of it, based on this PR), and after it's already merged to dotnet/runtime.  So I disagree it's of limited value.

> As for locally a build step would not catch inadvertent adds

I don't follow. If I'm locally writing something and include a cred, when I build/test before submitting the PR, it flags it.

Reply:
>  If we're going to run it in the private mirror, why wouldn't we also want it in the main repo?

If the question is why do we run it on public code - that isn't clear to me. Nevertheless, they do and the goal of these suppressions is that future passes are much easier. If there is value in running such a step regularly on already-public code, it probably should run in the validation rings where they validate source server links, etc, so it isn't a burden on the product build.

> If I'm locally writing something and include a cred, when I build/test before submitting the PR, it flags it, and I fix it before submitting the PR.

Not all files are in the build - eg., if I have a text file with a cred in it, then do `git commit -am ..` the build will not see it. That's what I mean by inadvertent add, that only a git hook would see.

Reply:
Very few direct check-ins are made to the private mirror. It makes no sense to me to run analyzers over that source that we don't also run as part of the public repo, where 99.999% of all commits come from. Doing that only causes additional dev work and headaches, and it's caused problems as well, where mirroring fails because the already merged commits then fail the subsequent pre-merge validation. This also isn't a one-time thing; PRs like this have been submitted at multiple times in the past in order to mop up from such analyzers that are being run late. And to your question about not deleting code as dead/stale, this exact mechanism would address that. So I don't understand the concerns or pushbavk. It seems like a no brainer to me. What am I not comprehending?

> Not all files are in the build

And what percentage of the changes in this PR are to such files? I'm not saying it would prevent all possible problems. You're saying it's useless if it's not 100% false negatives? I highly doubt the existing tooling even gets close to that.

Reply:
Just info about the tool and public PRs, it cannot be run out in the open: https://github.com/dotnet/arcade/issues/4663

But there's an open issue https://github.com/dotnet/core-eng/issues/5747 about running it on public PRs without disclosing the tool or the results.

Reply:
Thanks. I don't buy the answer in that issue, though. We run analyzers that validate, for example, SHA1 isn't being used, and that's an "SDL static verification" tool.

Reply:
@danmosemsft, are you ok to merge this PR? I think we currently don't have a better option how to suppress the bugs reported by the tool.

Reply:
The changes look OK to me, modulo the large conceptual problem of how there's a gap between checkins and analysis.  I'm withholding a checkmark to let @danmosemsft decide if we want to block this on the conceptual problem or not.